MetaMask/metamask-extension
# Adding popular custom network integration

## Explanation

Created a new feature for adding the popular custom networks. When enabled from the `Experimental` tab, this feature allows users to add one of the popular custom networks like Arbitrum One, Polygon Mainnet etc.

## More Information

Fixes: https://github.com/MetaMask/metamask-extension/issues/12903

## Screenshots/Screencaps

https://user-images.githubusercontent.com/92310504/175255384-cd9db04d-d608-4f9f-86be-5ac45eece225.mov

## Manual Testing Steps

1. Go to the `Settings`.
2. Click on the `Experimental` tab.
3. Enable the toggle switch `Show Custom Network List`.
4. From this step there are two ways to continue the flow:
    - The first one is to click on the `Networks` tab within the `Settings` and then to click on the button `Add a network`.
    - The second one is to open the network dropdown menu from the app header and then to click on the button `Add Network`.
5. The new screen is now shown. The modal view has the `back (<)` button which takes the user to the `Networks` screen. Also, the networks which rely on third parties have the exclamation triangle next to their name which can be hovered and then the tooltip with information shows up with a link `Learn more` which takes the user to the web-page `The risks of connecting to an unknown network`.
**NOTE**: If the button `Add a network manually` from the app footer is clicked, then the screen for manually adding the network is shown and from there the network can be added by entering the appropriate information about the network.
6.  The popular custom network can be added by clicking on the `Add` button. When clicked, the popup with the network information shows up. This popup has `tooltips (i)` which show information when hovered. Clicking on the `scams and network security risks` takes the user to the web-page `User Guide: Custom networks and sidechains`. If a button `View all details` is clicked then the new, smaller popup shows up with the information about the network and this popup can be closed by simply clicking on the `Close` button or `X`. 
7. The main popup can be closed by clicking on the `Cancel` button which rejects the pending approval for adding the network and the user stays on the same page. If the `Approve` button is clicked, then the network will be added, the user will be taken to the `Home page` and the new popup will show up.
8. From this step, user can switch to the newly created network by clicking on the button `Switch to network_name` or click on the `Dismiss` button and later switch to that network.
